homeassistant/components/sabnzbd/__init__.py

```diff
@@ -69,42 +96,79 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
+@callback
+def async_get_entry_id_for_service_call(hass: HomeAssistant, call: ServiceCall) -> str:
+    """Get the entry ID related to a service call (by device ID)."""
+    api_key = call.data[ATTR_API_KEY]
+
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.title == api_key:
+            return entry.entry_id
+
+    raise ValueError(f"No api for API key: {api_key}")
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the SabNzbd Component."""
     sab_api = await get_client(hass, entry.data)
     if not sab_api:
         raise ConfigEntryNotReady
 
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
```

Comment:
Please put this back below. It's more logical to forward the entry to the platforms after we're done with the setup here.

Reply:
Done